homeassistant/components/synology_dsm/common.py

```diff
@@ -225,6 +248,15 @@ def _setup_api_requests(self) -> None:
                 self.dsm.reset(self.security)
             self.security = None
 
+        if not self._with_file_station:
+            LOGGER.debug(
+                "Disable file station api from being updated or '%s'",
+                self._entry.unique_id,
+            )
+            if self.file_station:
+                self.dsm.reset(self.file_station)
+            self.file_station = None
```

Comment:
Looks like this is legitimately not covered

Reply:
yeah, unfortunately we've no integration setup tests at all, yet ... so to add a test for this might be not such easy :slightly_frowning_face: